demo/api/src/predefined-page/dto/predefined-page.input.ts

```diff
@@ -1,10 +1,12 @@
 import { Field, InputType } from "@nestjs/graphql";
-import { IsOptional, IsString } from "class-validator";
+import { IsEnum, IsOptional } from "class-validator";
+
+import { PredefinedPageType } from "../entities/predefined-page.entity";
 
 @InputType()
 export class PredefinedPageInput {
-    @Field(() => String, { nullable: true })
-    @IsString()
+    @Field(() => PredefinedPageType, { nullable: true })
+    @IsEnum(PredefinedPageType)
     @IsOptional()
```

Comment:
@nsams type can be both optional (not changed) and nullable (unset). Should I use `@IsUndefinable` and `@IsNullable()` instead? And should the TS type reflect that as well?

Reply:
yes, `@IsOptional` should not be used anymore. And in that case `@IsUndefinable` and `@IsNullable()` together are correct.

> And should the TS type reflect that as well?

yes